static/app/components/globalDrawer/index.tsx

```diff
@@ -0,0 +1,84 @@
+import {useCallback} from 'react';
+import styled from '@emotion/styled';
+
+import {closeDrawer as closeDrawerAction} from 'sentry/actionCreators/drawer';
+import {DrawerBody, DrawerPanel} from 'sentry/components/globalDrawer/components';
+import DrawerStore from 'sentry/stores/drawerStore';
+import {useLegacyStore} from 'sentry/stores/useLegacyStore';
+import {useEffectAfterFirstRender} from 'sentry/utils/useEffectAfterFirstRender';
+import {useLocation} from 'sentry/utils/useLocation';
+
+export interface DrawerOptions {
+  /** If true (default), closes the drawer when an escape key is pressed */
+  closeOnEscapeKeypress?: boolean;
+  /** If true (default), closes the drawer when anywhere else is clicked */
+  closeOnOutsideClick?: boolean;
+  /** Callback for when the drawer closes */
+  onClose?: () => void;
+  /** Callback for when the drawer opens */
+  onOpen?: () => void;
+}
+
+export interface DrawerRenderProps {
+  /** Body container for the drawer */
+  Body: typeof DrawerBody;
+  /** Close the drawer */
+  closeDrawer: () => void;
+}
```

Comment:
Please prefer `//` for comments like this. It makes it easier to comment whole chunks of code later without the extra `*/` getting in the way.

Also, vscode (if that's your editor) will insert/remove `//` if you type `CMD+/` on the current row(s)

Reply:
I get what you mean, I'm going to make these multi-line to make it more transparent but these are formatted to provide the IDE with descriptions of each key, not just info for those reading this specific file. Good catch!